### PR TITLE
Add a recipe for handling remote views

### PR DESCRIPTION
**Pull request by bentrengrove**
I could not find any documentation on handling remote views. Glide provides an `AppWidgetTarget` for updating widgets that I was looking for the equivalent of.

I thought about making a PR to add this as a target but saw that you are not accepting features at this time so I thought it would be good as a recipe. No problem if this is too niche but I think it could help others running in to the same problem.

Thanks for your work on this library!

### CODE REVIEW

File: docs/recipes.md [hunk 1]
```diff
@@ -197,3 +197,43 @@ detailImageView.load("https://www.example.com/image.jpg") {
...
+    }
+}
+```
+
+Then just set it as your target when enqueuing.
+```kotlin
+val remoteViewTarget = RemoteViewTarget(...)
+val request = ImageRequest.Builder(context)
+    .data("https://www.example.com/image.jpg")
+    .target(remoteViewTarget).build()
```
Comment (colinrtwhite) [inline docs/recipes.md:236]: Nit: Can you put `.build()` on the next line?

File: docs/recipes.md [hunk 2]
```diff
@@ -197,3 +197,43 @@ detailImageView.load("https://www.example.com/image.jpg") {
 - Use the [`MemoryCache.Key`](getting_started.md#memory-cache) of the start image as the [`placeholderMemoryCacheKey`](../api/coil-base/coil.request/-image-request/-builder/placeholder-memory-cache-key) for the end image. This ensures that the start image is used as the placeholder for the end image, which results in a smooth transition with no white flashes if the image is in the memory cache.
 
 - Use [`ChangeImageTransform`](https://developer.android.com/reference/android/transition/ChangeImageTransform) and [`ChangeBounds`](https://developer.android.com/reference/android/transition/ChangeBounds) together for optimal results.
+
+## Remote Views
+
+Coil does not provide a `RemoteViewTarget` out of the box. You can easily recreate one though for updating remote views like widgets.
+
+```kotlin
+class RemoteViewsTarget(private val context: Context, private val componentName: ComponentName, private val remoteViews: RemoteViews, @IdRes val imageViewId: Int) : Target {
```
Comment (colinrtwhite) [inline docs/recipes.md:206]: Nit: Please put the constructor params on separate lines.

**Comment by colinrtwhite** [comment]
This is great, thanks! Yep, I think I want to avoid having a remove views target in the core library since it's a nice use case, but it's a good idea to have a recipe for it.

**Comment by bentrengrove** [comment]
No problem, have fixed those formatting errors up.

**Review by colinrtwhite — APPROVED** [review]
Awesome, thanks!

**Pull request merged**